### DISCUSSION TITLE
How to retrieve the super constructor call with Konsist?

### DISCUSSION THREAD

**Discussion by vladislav-chetrari** [LemonAppDev/konsist#1079] [Ask a Question]
### Wanted end result.

I try to check whether specific parameters are passed to parent's constructor via [Konsist](https://github.com/LemonAppDev/konsist). For example: I have a ViewModel like:
```
class ExampleViewModel @Inject constructor(
  private val useCase1: SomeUseCase,
  private val useCase2: OtherUseCase,
) : BaseViewModel(useCase1, useCase2) { ... }
```

I want to check that all the useCases are passed to parent's constructor. Is there a way to get parent constructor call out of KoClassDeclaration?

### What you tried so far.

For now I just used a naive way to take KoClassDeclaration.text and substring until I get parameters names

**Comment by nataliapeterwas** [comment]
Konsist `0.17.0` has been released! This release includes this improvement :) 

The following test should be helpful in your case:
```kotlin
Konsist
   .scopeFromProject()
   .classes()
   .assertTrue {
      val parameters = it
         .primaryConstructor
         ?.parameters
         ?.map { it.name }

      val arguments = it
         .parentClass
         ?.arguments
         ?.map { it.value }

      parameters == arguments
   }
```

I closed this discussion. If anything is unclear or missing, feel free to start a new one.
